Neos.Flow/Classes/Mvc/Dispatcher.php

```diff
@@ -123,28 +123,29 @@ public function dispatch(ActionRequest $request): ActionResponse
     protected function initiateDispatchLoop(ActionRequest $request): ActionResponse
     {
         $dispatchLoopCount = 0;
-        while ($request->isDispatched() === false) {
-            if ($dispatchLoopCount++ > 99) {
-                throw new Exception\InfiniteLoopException(sprintf('Could not ultimately dispatch the request after %d iterations.', $dispatchLoopCount), 1217839467);
+        do {
+            if ($request === null) {
+                return $response ?? new ActionResponse();
             }
             $controller = $this->resolveController($request);
             $this->emitBeforeControllerInvocation($request, $controller);
             try {
                 $response = $controller->processRequest($request);
                 $this->emitAfterControllerInvocation($request, $response, $controller);
+                return $response;
             } catch (StopActionException $stopActionException) {
                 $this->emitAfterControllerInvocation($request, $stopActionException->response, $controller);
                 $response = $stopActionException->response;
-                if (!$request->isMainRequest()) {
-                    $request = $request->getParentRequest();
+                if ($request->isMainRequest()) {
+                    return $response;
                 }
+                $request = $request->getParentRequest();
```

Comment:
as discussed we concluded that this logic might be faulty and actually do harm without the `isDispatched` in place.

background:

the dispatcher is either used for main request, in which case this logic is not needed,
or for subrequest like in a fusion plugin

scenario:

a subrequest, while being handled, throws a `StopActionException`.
which is catched here. We check if its a main request and get the main request which is now to be handled in the loop, like a forward.

Thesis:

Now it will actually render the outer response inside the plugin
But previously because the main request was dispatched === true, it would not be handled.

So in theory this is dead code, and now must be removed.

Reply:
Yes the thesis seems to be correct. Using `throwStatus` in a plugin will result in the main request rendering being started again an again until the memory is exhausted.

Removing this code will actually make the tests `dispatchIgnoresStopExceptionsForFirstLevelActionRequests` and `dispatchCatchesStopExceptionOfActionRequestsAndRollsBackToTheParentRequest` fail, so i wanted to research about the original intentions.
It seems that with the introduction https://github.com/neos/flow-development-collection/commit/ff050845e3659c08189026e374334b579865e32c#diff-450585af3bb27be2be268192637242195c5d6308512fd53dac9fa248ff319686 of sub request, this logic was introduced. (perviously all first level `StopActionException` where silently swallowed)

```php
} catch (\F3\FLOW3\MVC\Exception\StopActionException $stopActionException) {
    if ($request instanceof \F3\FLOW3\MVC\Web\SubRequest && $request->isDispatched()) {
        throw $stopActionException;
    }
}
```

Reply:
i removed the 

```php
if (!$request->isMainRequest()) {
    $request = $request->getParentRequest();
}
```

part in https://github.com/neos/flow-development-collection/pull/3301/commits/0ed5f28a0e574882c38e065d1bb66e5e4c407d1f ;)